committer/src/main.rs

```diff
@@ -49,18 +49,32 @@ async fn main() -> Result<()> {
         commit_interval,
         ethereum_rpc.clone(),
         storage.clone(),
-        fuel_adapter,
+        fuel_adapter.clone(),
         &config,
         &metrics_registry,
         cancel_token.clone(),
     );
 
     let listener_handle = setup::l1_event_listener(
         &internal_config,
+        ethereum_rpc.clone(),
+        storage.clone(),
+        &metrics_registry,
+        cancel_token.clone(),
+    );
+
+    let state_committer_handle = setup::state_committer(
```

Comment:
How are we going to roll out this feature ? Would it make senses to have some cli flag to enable it until it is finalized ?